trin-core/src/portalnet/discovery.rs

```diff
@@ -151,11 +151,12 @@ impl Discovery {
     pub async fn send_talkreq(
         &self,
         enr: Enr,
+        protocol: &str,
```

Comment:
Minor nit: if you're converting to a `Vec` later anyways, you should take a `String`, which _can_ save a copy.